### PR TITLE
DatePicker Integration

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#6803

*Description of changes:*

* Removes localized DatePicker component.

* Integrates Shared DatePicker.

* **Fix to alteration build template, to add NR data ONLY after changes.**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/mixins/filing-template-mixin.ts [hunk 1]
```diff
@@ -252,6 +251,15 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
...
 
+    // Apply Name Request data if applicable
+    if (this.getNameRequest.nrNumber) {
+      filing.alteration.nameRequest = {
+        legalType: this.getNameRequest.legalType,
+        legalName: this.getNameRequest.legalName,
+        nrNumber: this.getNameRequest.nrNumber
+      }
+    }
+
```
Comment (jordiwes) [inline src/mixins/filing-template-mixin.ts:262]: I'm guessing this is part of a different ticket :)

Comment (cameron-freshworks) [reply src/mixins/filing-template-mixin.ts:262]: I'm not sure this is captured in another ticket, just that i noticed we were passing a lot of unwanted properties to the api in the event of company name change. This extra properties were causing the api to error.

Figured i'd fix it proactively before i saw a bug ticket 😁

Comment (jordiwes) [reply src/mixins/filing-template-mixin.ts:262]: Cool! got it

File: package.json [hunk 2]
```diff
@@ -19,6 +19,7 @@
     "@bcrs-shared-components/contact-info": "1.0.8",
     "@bcrs-shared-components/corp-type-module": "1.0.3",
     "@bcrs-shared-components/court-order-poa": "1.0.1",
+    "@bcrs-shared-components/date-picker": "1.0.0",
```
Comment (severinbeauvais) [inline package.json:22]: Certify component version (line 17) should be "1.0.12" per Steph's recent PR. Please rebase and make sure it's correct thereafter.

File: src/components/dialogs/ResolutionDateDialog.vue [hunk 3]
```diff
@@ -9,12 +9,14 @@
...
-                     @emitCancel="exit()"
-                     @emitDateSync="date = $event"/>
+        <date-picker
+          title="Resolution or Court Order Date"
+          :error-msg="errors"
+          nudge-right="100"
+          @emitDate="onDateEmitted($event)"
+          @emitCancel="exit()"
+          @emitDateSync="date = $event"
+        />
```
Comment (severinbeauvais) [inline src/components/dialogs/ResolutionDateDialog.vue:19]: Could you please confirm with Linda and if OK then add a maxDate (today) here? (This requirement came after my ticket was implemented.)

Thanks for updating this page BTW 👍

File: src/mixins/filing-template-mixin.ts (hunk 1 above)
Comment (severinbeauvais) [reply src/mixins/filing-template-mixin.ts:262]: Sounds good. Please call it out in the PR description in case of future post-mortem 😁

**Review by severinbeauvais — APPROVED** [review]
Looks good.

FYI - If you merge first then I'll update my PR, or vice-versa.

File: package.json (hunk 2 above)
Comment (cameron-freshworks) [reply package.json:22]: kk, will do

File: src/components/dialogs/ResolutionDateDialog.vue (hunk 3 above)
Comment (cameron-freshworks) [reply src/components/dialogs/ResolutionDateDialog.vue:19]: I believe there is was much more work to come here with regards to these dates, following up tho!

File: src/mixins/filing-template-mixin.ts (hunk 1 above)
Comment (cameron-freshworks) [reply src/mixins/filing-template-mixin.ts:262]: Done!

**Comment by cameron-freshworks** [comment]
I have an interesting build error in the CI which i see no relation too within the PR.. 
Any thoughts on this?

File: src/components/dialogs/ResolutionDateDialog.vue (hunk 3 above)
Comment (severinbeauvais) [reply src/components/dialogs/ResolutionDateDialog.vue:19]: Ah, ok. If it's not small then I suggest a new ticket for that.

**Comment by severinbeauvais** [comment]
> I have an interesting build error in the CI which i see no relation too within the PR..
> Any thoughts on this?

Oh shoot, that may be something I broke with my vue/cli-service change recently. Looking...

**Comment by cameron-freshworks** [comment]
> > I have an interesting build error in the CI which i see no relation too within the PR..
> > Any thoughts on this?
> 
> Oh shoot, that may be something I broke with my vue/cli-service change recently. Looking...

np! 
Looks like a dependency of `@vue/cli-service`

**Comment by severinbeauvais** [comment]
> Maybe we need to add `vue-loader-v16` as a dependency?

Ohhhhhhh. I know nothing about this. Obviously you researched this. Is it supposed to be a main or dev dependency?

**Comment by cameron-freshworks** [comment]
> > Maybe we need to add `vue-loader-v16` as a dependency?
> 
> Ohhhhhhh. I know nothing about this. Obviously you researched this. Is it supposed to be a main or dev dependency?

It's probably a dev dependency 🤔 
It's a requirement of the vue-cli

**Comment by severinbeauvais** [comment]
I just read up a bit on npm. It looks a bit hacky. Let's keep an eye on it and see if we can get rid of it some day.

File: src/components/dialogs/ResolutionDateDialog.vue (hunk 3 above)
Comment (cameron-freshworks) [reply src/components/dialogs/ResolutionDateDialog.vue:19]: ah yes, work is captured here: https://app.zenhub.com/workspaces/entity-5bf2f2164b5806bc2bf60531/issues/bcgov/entity/6904

Certainly covers a few things, so we can wait on applying them here.

Comment (severinbeauvais) [reply src/components/dialogs/ResolutionDateDialog.vue:19]: OK. Good to merge when ready then.

**Pull request merged**